internal/webhooks/cluster_test.go

```diff
@@ -1249,6 +1257,132 @@ func TestClusterTopologyValidation(t *testing.T) {
 					Build()).
 				Build(),
 		},
+		{
+			name:      "should update if cluster is fully upgraded and up to date",
+			expectErr: false,
+			old: builder.Cluster("fooboo", "cluster1").
+				WithControlPlane(builder.ControlPlane("fooboo", "cluster1-cp").Build()).
```

Comment:
Would be maybe better if we also set WithControlPlane in the second cluster (for all cases here)